File: front/client/src/app/Models/Course/CourseResponse.ts
import { Language } from "../enums/Language";
import { PrivateFileDataResponse } from "../File/PrivateFileDataResponse";
import { ParagraphResponse } from "./Paragraph/ParagraphResponse";

export interface CourseResponse {
    id: number;
    authorId: number;
    photo: PrivateFileDataResponse;
    video: PrivateFileDataResponse;
    name: string;
    price: number;
    isPublished: boolean;
    primaryLanguage: string;
    paragraphs: ParagraphResponse[];
}

File: src/Learnify/Learnify.Core/Dto/Course/CourseResponse.cs
using Learnify.Core.Dto.Course.ParagraphDtos;
using Learnify.Core.Dto.File;
using Learnify.Core.Enums;

namespace Learnify.Core.Dto.Course;

public class CourseResponse
{
    public int Id { get; set; }
    public int AuthorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string PrimaryLanguage { get; set; }
    public PrivateFileDataResponse Photo { get; set; }
    public PrivateFileDataResponse Video { get; set; }
    public IEnumerable<ParagraphResponse> Paragraphs { get; set; }
}